ee/tasks/test/subscriptions/test_subscriptions.py

```diff
@@ -65,6 +66,39 @@ def test_subscription_delivery_scheduling(
 
         assert mock_deliver_task.delay.mock_calls == [call(subscriptions[0].id), call(subscriptions[1].id)]
 
+    @patch("ee.tasks.subscriptions.deliver_subscription_report")
+    def test_does_not_schedule_subscription_if_item_is_deleted(
+        self,
+        mock_deliver_task: MagicMock,
+        mock_gen_assets: MagicMock,
+        mock_send_email: MagicMock,
+        mock_send_slack: MagicMock,
+    ) -> None:
+        self.insight.deleted = True
+        self.insight.save()
+        self.dashboard.deleted = True
+        self.dashboard.save()
+
+        create_subscription(
```

Comment:
it's a nit but should we subscribe before deleting? as a more realistic order of setup

Reply:
Good point

Reply:
Done - still passes but might save us somehow in the future 🤯